src/v2/Components/ArtworkFilter/ArtworkFilters/PriceRangeFilterNew.tsx

```diff
@@ -0,0 +1,211 @@
+import { FC, useEffect, useState, useMemo, FormEvent } from "react"
+import { Box, Flex, LabeledInput, Spacer, Text } from "@artsy/palette"
+import {
+  SelectedFiltersCountsLabels,
+  useArtworkFilterContext,
+} from "../ArtworkFilterContext"
+import styled from "styled-components"
+import { themeGet } from "@styled-system/theme-get"
+import { FilterExpandable } from "./FilterExpandable"
+import { isCustomValue } from "./Utils/isCustomValue"
+import { useFilterLabelCountByKey } from "../Utils/useFilterLabelCountByKey"
+import { Range, RANGE_DOT_SIZE } from "v2/Components/Range"
+import { debounce } from "lodash"
+
+// Disables arrows in numeric inputs
+export const NumericInput = styled(LabeledInput).attrs({ type: "number" })`
+  /* Chrome, Safari, Edge, Opera */
+  input::-webkit-outer-spin-button,
+  input::-webkit-inner-spin-button {
+    -webkit-appearance: none;
+    margin: 0;
+  }
+
+  /* Firefox */
+  input[type="number"] {
+    -moz-appearance: textfield;
+  }
+
+  /* HACK: Setting the font-size to a minimum 16px prevents iOS from zooming on focus */
+  /* This won't be necessary when upgraded to Palette v3 */
+  @media ${themeGet("mediaQueries.xs")} {
+    input {
+      font-size: 16px;
+    }
+  }
+`
+
+type CustomRange = (number | "*")[]
+
+// Constants
+const DEBOUNCE_DELAY = 300
+const DEFAULT_CUSTOM_RANGE: CustomRange = ["*", "*"]
+const DEFAULT_PRICE_RANGE = "*-*"
+const DEFAULT_RANGE = [0, 50000]
+
+export const parseRange = (range: string = DEFAULT_PRICE_RANGE) => {
+  return range.split("-").map(s => {
+    if (s === "*") return s
+    return parseInt(s, 10)
+  })
+}
+
+const parseSliderRange = (range: CustomRange) => {
+  return range.map((value, index) => {
+    if (value === "*") {
+      return DEFAULT_RANGE[index]
+    }
+
+    return value as number
+  })
+}
+
+export const convertToFilterFormatRange = (range: number[]) => {
+  return range.map((value, index) => {
+    if (value === DEFAULT_RANGE[index]) {
+      return "*"
+    }
+
+    return value
+  })
+}
+
+export const getValue = (value: CustomRange[number]) => {
+  return value === "*" || value === 0 ? "" : value
+}
+
+export interface PriceRangeFilterNewProps {
+  expanded?: boolean
+}
+
+export const PriceRangeFilterNew: FC<PriceRangeFilterNewProps> = ({
+  expanded,
+}) => {
+  const {
+    shouldStageFilterChanges,
+    currentlySelectedFilters,
+    setFilter,
+  } = useArtworkFilterContext()
+  const { priceRange, reset } = currentlySelectedFilters?.() ?? {}
+  const [range, setRange] = useState(parseRange(priceRange))
+  const sliderRange = parseSliderRange(range)
+  const [minValue, maxValue] = range
+  const [defaultMinValue, defaultMaxValue] = DEFAULT_RANGE
+
+  const filtersCount = useFilterLabelCountByKey(
+    SelectedFiltersCountsLabels.priceRange
+  )
+  const label = `Price${filtersCount}`
+  const selection = currentlySelectedFilters?.().priceRange
```

Comment:
Since we're already calling this above do we need to call it again? 

(side-thought: It might be worth quickly refactoring this `currentlySelectedFilters` behind a `useCurrentlySelectedFilters` hook so we don't always have to call `useArtworkFilterContext` to get at this commonly used function)

Reply:
@damassi do you mind if these changes are added in a separate PR?